develop/vagrant/README.md

```diff
@@ -21,24 +34,181 @@ It is recommended to then install the following VSCode extensions if you don't a
+
+### 6. Re-provisioning vagrant 
+At some points during development it may be desirable to re-run all the provisioners, i.e. after a change to the vagrant scripts has added new components or updated the versions of existing components
+
+re-running provisioners is possible by running
+```
+vagrant reload --provision
+
+```
+
+### 7. Build issues you may encounter for vagrant:
+The version of vagrant-libvirt that had been installed on my system when I used "sudo apt install vagrant" was old and it failed to update it automagically.
+so I had to
+
+```
+sudo apt remove vagrant-libvirt
+```
+After that, use following command to install vagrant libvirt plugin
+
+```
+vagrant plugin install vagrant-libvirt
+```
+Hit an error caused by unknown configuration section 'disksize'
+
+There are errors in the configuration of this machine. Please fix
+the following errors and try again:
+  
+Vagrant:
+* Unknown configuration section 'disksize'.
+I manually installed the plugin for disksize
+
 ```
-### Rebuilding Dsm
+vagrant plugin install vagrant-disksize
 ```
-cd ~/srcDsm/rdk-smart-router/dsm/build/
+If your Linux flavour has Secure Boot enabled, you will be required to sign the modules of the VM libs of the provider. You can use the following command to do so
+```
+sudo /usr/src/linux-headers-$(uname -r)/scripts/sign-file sha256 ./MOK.priv ./MOK.der $(modinfo -n MODNAME)
+```
+MOK.priv and MOK.der are the Machine Owner Keys you have to generate for yourself and enroll into the system. The sign-file should be shipped with your distribution or you should be able to download it in one of the packages for your distro.
+
+## 8. Building Dsm inside vagrant.
+```
+cd srcDsm/DSM/src
+mkdir build
+cde build
+  * build without rbus
+    cmake ../..
+* build  rbus
+    cmake ../.. -D ENABLE_RBUS_PROVIDER=ON
+make
+sudo make install
+```
+
+## 9. Building Dobby inside vagrant.
+```
+cd ~/srcDobby/build
+cmake ../
 make
 sudo make install
 ```
 
-It is possible to add build-time arguments to CMake when using the "Build" button in VSCode (e.g. to make sure the platform is set to DEV_VM). Open/create the file .vscode/settings.
-json and add the following JSON section:
-```json
-"cmake.configureArgs": [
-    "-DRDK_PLATFORM=DEV_VM"
-]
+### 11. Setting up a server with python to host containers
 ```
+sudo python3 -m http.server 8080
+
+```
+
+### 12. Clean up any installed containers.
+```
+rm -rf ~/destination/*
+
+```
+
+
+### 10. Run inside vagrant DSM with Dobby and rbus
+```
+vagrant ssh
+sudo DobbyDaemon --nofork
+with rbus we also need
+vagrant ssh
+killall -9 rtrouted;rm -fr /tmp/rtroute*;rtrouted -f -l DEBUG
+```
+In he next terminal start dsm:
+```
+vagrant ssh
+dsm
+
+```
+
+
+### 11. Deployment containers via rbus:
+
+* DSM rbus provider implements installDU() with URL (mandatory) and name (optional default is "default")
+
+
+* build and install DSM
```

Comment:
this step is already done by the vagrant file, so shouldn't have to be done again here

Reply:
Addressed.